datalad/distribution/install.py

```diff
@@ -207,6 +207,14 @@ def __call__(
             ds = require_dataset(dataset, check_installed=True,
                                  purpose='installation')
             common_kwargs['dataset'] = dataset
+        else:
```

Comment:
according to codecov coloring this codepath is not tested

Reply:
Yes. Test needs to follow. Just wasn't clear what really is the thing we want. What I pushed here is already the third iteration of my exploration yesterday ;-)